evalml/data_checks/invalid_targets_data_check.py

```diff
@@ -134,13 +134,6 @@ def validate(self, X, y):
                     message_code=DataCheckMessageCode.TARGET_MULTICLASS_HIGH_UNIQUE_CLASS,
                     details=details).to_dict())
 
-        if len(value_counts) == 2 and is_supported_type:
```

Comment:
Looks good! I agree we should keep the `TARGET_BINARY_INVALID_VALUES` code since we'll probably need to use it in #1954